### ISSUE TITLE
Enhance configuration to specify how a transformation is triggered. 

### ISSUE THREAD

**Issue by clementval** [C2SM-RCM/claw-compiler#194]
Currently, transformation are triggered by directive or hardcoded when they have to be triggered for each translation unit. 

The idea would be to set this information in the configuration file. 

```xml
<claw version="0.3">
  <global>
  </global>
  <groups>
    <group type="independent" name="internal-xcodeml" trigger="translation_unit" class="cx2x.translator.transformation.utility.XcodeMLWorkaround" />
    <group type="independent" name="parallelize-forward" trigger="directive" class="cx2x.translator.transformation.claw.parallelize.ParallelizeForward" />
  </groups>
</claw>
```

**Comment by clementval** [comment]
Implemented by #195
